frontend/src/pages/Graph/SummaryPanelNode.tsx

```diff
@@ -369,12 +370,18 @@ export class SummaryPanelNodeComponent extends React.Component<SummaryPanelNodeC
             <span style={{ paddingLeft: '4px' }}>Has Mirroring</span>
           </div>
         )}
-        {hasMissingSC && (
+        {isOutOfMesh && !serverConfig.ambientEnabled && (
```

Comment:
I don't think you need the `serverConfig.ambientEnabled`.  Instead, I think you can use the `nodeData.isAmbient` setting which you already added to GraphNodeData.

Reply:
But in that case, we can just check `isOutOfMesh`, because `isOutOfMesh` checks that the node meets both requirements -`!isAmbient` and !sidecars -, so in this case this comparison is not needed. 

The use of the global ambientEnabled is whether we change the message for nodes out of the mesh, from "Missing sidecars" to "Out of Mesh"

Reply:
> The use of the global ambientEnabled is whether we change the message for nodes out of the mesh, from "Missing sidecars" to "Out of Mesh"

The global setting is for Istio overall, all namespaces, right?  But the node's `isAmbient` is specific to the node, so why not use `node.isAmbient` to make this decision.  Maybe I am confused :-)

Reply:
Yes, that's right. 

I can put an example, 

![image](https://github.com/kiali/kiali/assets/49480155/de8ee39d-eddb-4ee8-8301-c60306e049b2)
 
This namespace doesn't have any annotation. Is this the message that we want to show even when we start the Mesh with Ambient profile? I think if we are in Ambient profile we should show "Out of Mesh". But if we check node.isAmbient it would be false in this case, and the message would be "Has missing sidecar". 

Checking globally if Ambient is enabled, when the node is out of mesh (Because the node has no sidecars or Ambient annotations) we can show the message "Out of Mesh". 

But maybe, this is not the desired behavior?

Reply:
Installing with the Ambient profile means that namespaces in the mesh may be, but are not required to be, enabled for Ambient, right?   In the example above, the `local-path-storage` namespace is not annotated for Ambient.  The `local-path-provisioner` I think would have `node.isOutOfMesh: true` and `node.isAmbient: false`.  And so I think it is correct to show `Has Missing Sidecar` for the OutOfMesh situation.  If `local-path-storage` were annotated for Ambient,  then I think would have `node.isOutOfMesh: true` and `node.isAmbient: true`.  And so the message would be something like, "Out of Ambient Mesh".  Maybe this is the confusion.  I was expecting that `node.isAmbient` would be true for all nodes in an Ambient namespace.  But perhaps you expect, that, if the workload somehow opted out of the ambient mesh, it would be `node.isAmbient: false`.   That may be a fair assumption, but in that case I don't know how `isAmbient` is helpful, it is just duplicating `isOutOfMesh`.   In the way I am thinking `isAmbient` is just a convenience, because you could also remove that attribute and check the same thing with a utility function like, `isAmbient(node.namespace)`.

Reply:
Ah so, I think the confusion was about the expected behavior. In the case of `local-path-store` namespace if it is not annotated for Ambient, but does not have the injection enabled annotation, I wouldn't expect to be warned about sidecars, because you have different elements to include the node in the mesh, so why just the sidecars are a warning? (In case you are in Ambient profile)

It is how it works at the moment in other pages: 

![image](https://github.com/kiali/kiali/assets/49480155/74173b04-f192-4aa6-917a-1be032adcbe4)

![image](https://github.com/kiali/kiali/assets/49480155/a873f61c-2fad-4788-9fce-0ea9453a4420)

In case you want to show the `missing sidecars` message, we could use `node.isAmbient` in the way you have mentioned, then.

Reply:
> I wouldn't expect to be warned about sidecars, because you have different elements to include the node in the mesh, so why just the sidecars are a warning? (In case you are in Ambient profile)

I think I am just making this more confusing.  I think we both want the same thing.  My only question was whether it was needed to use the global ambient profile setting in this logic.  If you think it is necessary then it is fine for me.  I do wonder whether the new `node.isAmbient` is needed.  Maybe it could be removed, and just have `node.isOutOfMesh`.  If we need to check whether the node is in an ambient namespace then we could have a utility function like `isAmbient(node.namespace)`.

Reply:
Yes, the node isAmbient property was used in some other place but I have changed to use the global ambient profile to keep consistency with the rest of the application Ambient messages.